src/js/files/components/UploadOverlay.js

```diff
@@ -36,15 +38,54 @@ const UploadOverlayList = styled(BoxGroupSection)`
     padding: 0;
 `;
 
+const UploadInfo = styled.div`
+    padding-bottom: 5px;
+`;
+
+const FormattedRemainingUploadTime = styled.div`
+    display: flex;
+    justify-content: space-between;
+    font-size: ${getFontSize("md")};
+`;
+
+const UploadInformation = ({ remaining, uploadSpeed }) => {
+    const remainingTime = remaining ? remaining : 3600;
+    const estimatedUploadSpeed = uploadSpeed ? (uploadSpeed / 1000000).toFixed(0) : 0;
+
+    let formattedTimeRemaining = "";
+    if (secondsToMinutes(remainingTime) === 0) {
+        formattedTimeRemaining = `${remainingTime.toFixed(0)} seconds remaining`;
+    } else if (secondsToMinutes(remainingTime) > 0 && secondsToMinutes(remainingTime) < 60) {
+        formattedTimeRemaining = `${secondsToMinutes(remainingTime)} minute(s) remaining`;
+    } else if (secondsToHours(remainingTime) > 0 && secondsToHours(remainingTime) <= 12) {
+        formattedTimeRemaining = `${secondsToHours(remainingTime)} hour(s) remaining`;
+    } else {
+        formattedTimeRemaining = " > 12hr remaining";
+    }
```

Comment:
Can this be taken care of by [`formatDuration`](https://date-fns.org/v2.28.0/docs/formatDuration)?

Reply:
yes